trade-finance-manager-api/src/v1/controllers/amendment.controller-updateFacilityAmendment.test.js

```diff
@@ -0,0 +1,141 @@
+
+const { underwriter, underwriterManager } = MOCK_USERS_FOR_TASKS;
+
+describe('controllers - amendment, function updateFacilityAmendment', () => {
+  let getTasksAssignedToUserByGroupMock;
+
+  const { amendmentId, facilityId } = MOCK_AMENDMENT;
+
+  const facility = {
+    _id: facilityId,
+    facilitySnapshot: { ...MOCK_FACILITIES[0], _id: facilityId },
+    tfm: {},
+  };
+  const deal = {
+    dealSnapshot: MOCK_DEAL,
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    api.findUserById.mockReset();
+    mockFindUserById();
+
+    jest.useFakeTimers();
+    jest.setSystemTime(MOCK_TASKS_CHANGE_TIME * 1000); // Tuesday, December 23, 2008 2:40:00 AM GMT
+
+    // Mock setup is minimal, just to reach function getTasksAssignedToUserByGroup.
+    api.getAmendmentById = jest.fn().mockResolvedValue(MOCK_AMENDMENT);
+    api.updateFacilityAmendment = jest.fn(() => Promise.resolve(MOCK_AMENDMENT));
+    api.findOneFacility = jest.fn().mockResolvedValue(facility);
+    api.findOneDeal = jest.fn().mockResolvedValue(deal);
+  });
+
+  describe('PUT - updateFacilityAmendment', () => {
+    const requestBody = {
+      leadUnderwriter: {
+        _id: underwriter._id,
+        firstName: underwriter.firstName,
+        lastName: underwriter.lastName,
+      },
+    };
+
+    it('should call getTasksAssignedToUserByGroup if request contains leadUnderwriter', async () => {
+      // Arrange
+      const { req, res } = createMocks({ params: { amendmentId, facilityId }, user: underwriter, body: requestBody });
+
+      // Act
+      await amendmentController.updateFacilityAmendment(req, res);
+
+      // Assert
+      expect(createTasksAmendmentHelper.getTasksAssignedToUserByGroup).toHaveBeenCalledTimes(1);
+      expect(createTasksAmendmentHelper.getTasksAssignedToUserByGroup).toHaveBeenCalledWith(
+        MOCK_AMENDMENT.tasks,
+        TEAMS.UNDERWRITERS.id,
+        requestBody.leadUnderwriter._id,
+      );
+    });
+
+    it("should not call getTasksAssignedToUserByGroup if request doesn't contains leadUnderwriter", async () => {
+      // Arrange
+      const { req, res } = createMocks({ params: { amendmentId, facilityId }, user: underwriter, body: {} });
+
+      // Act
+      await amendmentController.updateFacilityAmendment(req, res);
+
+      // Assert
+      expect(createTasksAmendmentHelper.getTasksAssignedToUserByGroup).not.toHaveBeenCalled();
+    });
+
+    it('should call api.updateFacilityAmendment with tasks returned by getTasksAssignedToUserByGroup for Underwriter', async () => {
+      // Arrange
+      const { req, res } = createMocks({ params: { amendmentId, facilityId }, user: underwriter, body: requestBody });
+
+      getTasksAssignedToUserByGroupMock = jest.spyOn(createTasksAmendmentHelper, 'getTasksAssignedToUserByGroup');
+      getTasksAssignedToUserByGroupMock.mockResolvedValue(TASKS_ASSIGNED_TO_UNDERWRITER);
+
+      const auditDetails = { id: underwriter._id, userType: 'tfm' };
```

Comment:
Can `tfm` be a constant please?

Reply:
'tfm' string is used a lot in auditDetails functionality, introducing constant would be quite big touch to auditDetails code. This PR is not related to auditDetails, so lets leave 'tfs' as a string? 

https://github.com/UK-Export-Finance/dtfs2/blob/main/libs/common/src/types/audit-details.ts